app/workers/segment_delete_users_worker.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class SegmentDeleteUsersWorker < ApplicationJob
+
+  queue_as :low
+
+  # :reek:FeatureEnvy can be ignored here for the 'response'
+  def perform
+    return unless Features::SegmentDeletionConfig.enabled?
+    config = Features::SegmentDeletionConfig.config
+    DeletedObject.users.select(:id, :object_id).order(:id).find_in_batches(batch_size: config.request_size) do |records|
+      Segment::DeleteUsersService.call(records.map(&:object_id))
+      records.each(&DeleteObjectHierarchyWorker.method(:perform_later))
+      sleep(config.wait_time)
+    end
+  rescue Segment::ClientError, Segment::UnexpectedResponseError => error
```

Comment:
Should we abort the entire job because one batch had one failed request? 🤔

Reply:
If it is ClientError probably so because it means that we are doing something wrong, either wrong url, or RateLimitExceeded or something like that, and it will be the same for all of them 🤔

Reply:
I have done it the following way:
- If it fails for ServerError, it will retry, and it will continue where it left it for the users that were left to work with, because it has removed the records that has already worked on, and ONLY those, because if it fails, it does not reach the next line.
- If it fails for ClientError, it is our fault, and we don't want it to keep retrying for other users because it will keep failing.

WDYT of what it should do @guicassolato @duduribeiro @hallelujah @macejmic ?

Reply:
yeah, that makes sense for me..